lib/http/response/body.rb

```diff
@@ -53,7 +53,10 @@ def to_s
           @streaming  = false
           @contents   = String.new("").force_encoding(encoding)
 
-          while (chunk = @stream.readpartial)
+          length = @length || Float::INFINITY
```

Comment:
Yes, go ahead. I'm fine with keyword args. And it won't take long this time, I promise - will merge as soon as it's ready :D before this weekend.